src/scripts/main.js

```diff
@@ -1,3 +1,243 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const tbody = document.querySelector('table tbody');
+const wrapperForm = document.createElement('div');
+
+wrapperForm.style.cssText = `display: flex; flex-direction: column; row-gap: 20px; position: relative`;
+
+// --------------------SORTING OF TABLE
+table.addEventListener('click', (e) => {
+  const target = e.target.closest('thead th');
+
+  if (!target) {
+    return;
+  }
+
+  setAttr(target);
+
+  sort(target);
+});
+
+// CHANGE STYLES FOR CURRENT ROW
+table.addEventListener('click', (e) => {
+  const target = e.target.closest('tbody tr');
+
+  if (!target) {
+    return;
+  }
+
+  [...tbody.rows].forEach((el) => {
+    el.classList.remove('active');
+  });
+
+  target.classList.add('active');
+});
+
+// ---------------------ATTRIBUTE SETTING TO CONTROL SORTING
+function setAttr(elem) {
+  let currentAttr = null;
+
+  if (!elem.hasAttribute('data-sort')) {
+    currentAttr = 'asc';
+  } else {
+    currentAttr = elem.dataset.sort === 'asc' ? 'desc' : 'asc';
+  }
+
+  const thead = [...table.rows[0].cells];
+
+  thead.forEach((el) => {
+    el.removeAttribute('data-sort');
+  });
+
+  elem.setAttribute('data-sort', currentAttr);
+}
+
+//  -------------------------SORTING ELEMENTS
+function sort(elem) {
+  // WE FIND THE COLUMN INDEX NUMBER
+  const thead = [...table.rows[0].cells];
+  let currentPos = null;
+
+  if (thead.indexOf(elem) >= 0) {
+    currentPos = thead.indexOf(elem);
+  }
+
+  const tableBody = [...table.rows].slice(1, -1).sort((a, b) => {
+    let firstValue = a.cells[currentPos].innerHTML;
+    let secondValue = b.cells[currentPos].innerHTML;
+
+    // WE CHECK WHETHER THESE ARE NUMBERS
+    if (!isNaN(parseFloat(firstValue)) || firstValue.includes('$')) {
```

Comment:
The condition `!isNaN(parseFloat(firstValue)) || firstValue.includes('$')` might not work as intended. If the value includes '$', it should be checked separately to ensure correct parsing and sorting.